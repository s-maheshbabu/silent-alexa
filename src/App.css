/* TODO The App related CSS should be deleted */
.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

/** Scaffolding **/
#page {
  display: grid;
  grid-template-columns: 1fr 300px;
  grid-template-rows: 30px auto 30px;
  height: 100vh;
}

/** Panel Module **/

/** TODO: Right panel should also be moved into its own CSS file once the right panel component gets created. **/
#rightpanel {
  display: grid;
  grid-template-rows: auto;
  height: calc(100vh - 60px);
}

.panel-body {
  overflow: auto;
}

/** Header **/
#header {
  grid-row: 1;
  grid-column: 1 / 3;

  display: grid;
  grid-template-columns: 1fr auto;
}

#header-info {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;

  display: flex;
  justify-content: center;
}

#header-controls {
  display: flex;
  height: 100%;
  justify-content: flex-end;
}

/** Footer **/

#footer {
  grid-column: 1 / 3;
  grid-row: 3;

  display: grid;
  grid-template-columns: 1fr auto;
}

#footer-info {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

#footer-controls {
  display: flex;
  height: 100%;
  justify-content: flex-end;
}

#footer-controls a {
  height: 100%;
  display: flex;
  align-items: center;
}

/** Styles **/
body {
  margin: 0;
  padding: 0;

  background-color: #d7dae0;
  font-family: -apple-system, system-ui, BlinkMacSystemFont, "Segoe UI", Roboto,
    "Helvetica Neue", Arial, sans-serif;
  font-size: 0.8rem;
  color: #2c313a;
}

#header {
  align-items: center;
  border-top: 1px solid #c5cad3;
}

#footer {
  align-items: center;
  border-top: 1px solid #c5cad3;
}

#footer-controls a {
  padding: 0 10px;
  color: initial;
  text-decoration: none;
}

#footer-controls a:hover {
  background-color: #c5cad3;
}
