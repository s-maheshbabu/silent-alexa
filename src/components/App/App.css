/** Scaffolding **/
#page {
  display: grid;
  grid-template-columns: 1fr 300px;
  grid-template-rows: 60px auto 30px;
  height: 100vh;
}

/** Styles **/
body {
  margin: 0;
  padding: 0;

  background-color: #ffffff;
  font-family: -apple-system, system-ui, BlinkMacSystemFont, "Segoe UI", Roboto,
    "Helvetica Neue", Arial, sans-serif;
  font-size: 0.8rem;
  color: #2c313a;
}
