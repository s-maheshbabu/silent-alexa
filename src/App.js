import React, { Component } from "react";
import "./App.css";

import Header from "./Header";
import Body from "./Body";
import Footer from "./Footer";
import AuthenticationInfo from "./AuthenticationInfo";

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      authenticationInfo: undefined
    };
  }
  render() {
    return (
      <div id="page">
        <Header
          updateAuthenticationInfo={authenticationInfo =>
            this.updateAuthenticationInfo(authenticationInfo)
          }
        />
        <Body authenticationInfo={this.state.authenticationInfo} />
        <Footer />
      </div>
    );
  }

  /**
   * Update the state only if authenticationInfo is defined
   */
  updateAuthenticationInfo = function(authenticationInfo) {
    if (
      authenticationInfo &&
      authenticationInfo instanceof AuthenticationInfo
    ) {
      this.setState({ authenticationInfo: authenticationInfo });
    }
  };
}

export default App;
