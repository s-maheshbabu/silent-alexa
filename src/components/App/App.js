import React, { Component } from "react";
import "./App.css";

import Header from "Header/Header";
import Body from "Body/Body";
import Footer from "Footer/Footer";

class App extends Component {
  render() {
    return (
      <div id="page">
        <Header />
        <Body />
        <Footer />
      </div>
    );
  }
}

export default App;
