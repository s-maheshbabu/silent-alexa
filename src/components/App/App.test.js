import React from "react";
import { shallow } from "enzyme";
import App from "./App";

it("renders correctly without crashing", () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();

  wrapper.unmount();
});
