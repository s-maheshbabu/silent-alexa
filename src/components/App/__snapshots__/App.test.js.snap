// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly without crashing 1`] = `
<div
  id="page"
>
  <Routes
    updateAuthenticationInfo={[Function]}
  />
  <Header
    clearAuthenticationInfo={[Function]}
    isAuthenticationInfoValid={[Function]}
  />
  <Body
    clearAuthenticationInfo={[Function]}
    isAuthenticationInfoValid={[Function]}
  />
  <Footer />
</div>
`;
